### ISSUE TITLE
Azure: Use Azure Service Operator (ASO) for cloud resource management

### ISSUE THREAD

**Issue by surajssd** [confidential-containers/cloud-api-adaptor#1224]
We have seen quite a few issues trying to write code in go sdk and not able to reconcile the clean up of resources. I propose that we start moving the Azure implementation to utilize the Azure Service Operator for better reliability. That way we reuse the work that a lot of tooling is using today, reduce errors, improve reliability. CAA just creates resources declaratively and ASO takes care of ensuring they are created.

https://azure.github.io/azure-service-operator/

**Comment by bpradipt** [comment]
Let's discuss this in the community meeting to understand how it interacts with the dangling resources cleanup implemented via peerpod-ctrl and other nuances. It might be helpful to have a high level overview of how the integration will look like.

**Comment by bpradipt** [comment]
AWS has https://github.com/aws-controllers-k8s

**Comment by bpradipt** [comment]
So to summarise we have the following options to offload VM lifecycle management

1. AWS - https://github.com/aws-controllers-k8s
2. Azure - https://azure.github.io/azure-service-operator/ 
3. GCP -  https://github.com/GoogleCloudPlatform/k8s-config-connector
4. Local (KVM) - https://github.com/kubevirt
5. Cross cloud - https://docs.crossplane.io/ 

This came up in community [slack discussion](https://cloud-native.slack.com/archives/C04A2EJ70BX/p1740106274391259) and it's time to revisit this and work on this

**Comment by tarumizu** [comment]
> So to summarise we have the following options to offload VM lifecycle management

Hello,

I am interested in this approach to offload VM lifecycle management to external specialized components, particularly focusing on Local (KVM) and KubeVirt. It could enhance reliability, which is especially beneficial for k8s administrators.

I imagine that the mechanism for offloading VM lifecycle management externally requires not only cloud provider-specific code but also modifications to the common logic. For example, it may be necessary to modify the current common logic for creating and deleting VM instances, transitioning it to a system that handles declarative resources like CRDs.

Is my understanding correct? If so, I would appreciate any updates or information regarding the latest developments in this area, if available.

**Comment by bpradipt** [comment]
> > So to summarise we have the following options to offload VM lifecycle management
> 
> Hello,
> 
> I am interested in this approach to offload VM lifecycle management to external specialized components, particularly focusing on Local (KVM) and KubeVirt. It could enhance reliability, which is especially beneficial for k8s administrators.
> 
> I imagine that the mechanism for offloading VM lifecycle management externally requires not only cloud provider-specific code but also modifications to the common logic. For example, it may be necessary to modify the current common logic for creating and deleting VM instances, transitioning it to a system that handles declarative resources like CRDs.
> 
> Is my understanding correct? If so, I would appreciate any updates or information regarding the latest developments in this area, if available.

You are right. I'm not aware of anyone in the community doing this work. However there is a related work that will help  - https://github.com/confidential-containers/cloud-api-adaptor/issues/1317
